stable_baselines/common/segment_tree.py

```diff
@@ -98,33 +112,46 @@ def sum(self, start=0, end=None):
     def find_prefixsum_idx(self, prefixsum):
         """
         Find the highest index `i` in the array such that
-            sum(arr[0] + arr[1] + ... + arr[i - i]) <= prefixsum
+            sum(arr[0] + arr[1] + ... + arr[i - i]) <= prefixsum for each entry in prefixsum
 
         if array values are probabilities, this function
         allows to sample indexes according to the discrete
         probability efficiently.
 
-        :param prefixsum: (float) upperbound on the sum of array prefix
-        :return: (int) highest index satisfying the prefixsum constraint
+        :param prefixsum: (numpy float) upper bounds on the sum of array prefix
+        :return: (numpy int) highest indexes satisfying the prefixsum constraint
         """
-        assert 0 <= prefixsum <= self.sum() + 1e-5
-        idx = 1
-        while idx < self._capacity:  # while non-leaf
-            if self._value[2 * idx] > prefixsum:
-                idx = 2 * idx
-            else:
-                prefixsum -= self._value[2 * idx]
-                idx = 2 * idx + 1
+        if isinstance(prefixsum, float):
+            prefixsum = np.array([prefixsum])
+        assert 0 <= np.min(prefixsum)
+        assert np.max(prefixsum) <= self.sum() + 1e-5
+        assert isinstance(prefixsum[0], float)
+
+        idx = np.zeros(len(prefixsum), dtype=int)+1
```

Comment:
PEP8: Whitespaces around operators (should be `idx = np.zeros(len(prefixsum), dtype=int) + 1`). Same appears elsewhere.